services/setup_topo_file.sh

```diff
@@ -0,0 +1,44 @@
+#!/bin/bash
+set -ex
+
+## Systemd service for setting up topo file for select GPU SKUs
+cat <<EOF >/usr/sbin/setup_topo.sh
+#!/bin/bash
+
+metadata_endpoint="http://169.254.169.254/metadata/instance?api-version=2019-06-04"
+vmSize=\$(curl -H Metadata:true \$metadata_endpoint | jq -r ".compute.vmSize")
+nccl_conf="/etc/nccl.conf"
+
+case \$vmSize in
+    Standard_NC*ads_A100_v4)
+        echo "NCCL_TOPO_FILE=/opt/microsoft/ncv4/topo.xml" >> \$nccl_conf;;
+    
+    Standard_ND96*v4)
+        echo "NCCL_TOPO_FILE=/opt/microsoft/ndv4/topo.xml" >> \$nccl_conf;;
+        
+    Standard_ND40rs_v2)
+        echo "NCCL_TOPO_FILE=/opt/microsoft/ndv2/topo.xml" >> \$nccl_conf;;
+
+    *) echo "No Topo file for \$vmSize";;
+esac
+EOF
+chmod 755 /usr/sbin/setup_topo.sh
+
+cat <<EOF >/etc/systemd/system/setup_topo.service
+[Unit]
+Description=Setup appropriate topo file for the SKU
+After=network.target
+
+[Service]
+Type=oneshot
+ExecStart=/usr/sbin/setup_topo.sh
+RemainAfterExit=true
+StandardOutput=journal
+
+[Install]
+WantedBy=multi-user.target
+EOF
+
+systemctl enable setup_topo
+systemctl start setup_topo
+systemctl is-active --quiet setup_topo
```

Comment:
instead of having two different services, can we just have one service for any SKU specific customizations?

Reply:
I thought maybe in future if we have more customizations, it'd bloat the file. But I guess having one file would make sense for now